chain/network/src/peer_manager/testonly.rs

```diff
@@ -365,12 +365,12 @@ impl ActorHandler {
         self.with_state(move |s| async move { s.peer_store.update(&clock) }).await;
     }
 
-    pub async fn send_ping(&self, nonce: u64, target: PeerId) {
-        self.actix
-            .addr
-            .send(PeerManagerMessageRequest::PingTo { nonce, target }.with_span_context())
-            .await
-            .unwrap();
+    pub async fn send_ping(&self, clock: &time::Clock, nonce: u64, target: PeerId) {
+        let clock = clock.clone();
+        self.with_state(move |s| async move {
```

Comment:
Oh this is an interesting function. (btw the function's impl seems to be slightly improvable by using BoxFuture and .boxed() instead of spelling out the type yourself and pinning yourself.

Reply:
ack